pkg/operator/baremetal_config.go

```diff
@@ -76,6 +84,31 @@ func getBaremetalProvisioningConfig(dc dynamic.Interface, configName string) (Ba
 		glog.Errorf("provisioningOSDownloadURL not found in Baremetal provisioning CR %s", configName)
 		return BaremetalProvisioningConfig{}, err
 	}
+	// If provisioningNetwork is not provided, set it to a default of "managed".
+	provisioningNetwork, foundNetwork, err := unstructured.NestedString(provisioningSpec, "provisioningNetwork")
+	if !foundNetwork {
+		glog.Infof("provisioningNetwork not found in Baremetal provisioning CR %s", configName)
+		if !provisioningDHCPExternal {
+			provisioningNetwork = provisioningNetworkManaged
+		} else if provisioningDHCPExternal {
+			provisioningNetwork = provisioningNetworkUnmanaged
+		}
+	}
+	if err != nil {
+		glog.Errorf("provisioningNetwork not found in Baremetal provisioning CR %s", configName)
+		return BaremetalProvisioningConfig{}, err
+	}
+	provisioningNetworkCIDR, found, err := unstructured.NestedString(provisioningSpec, "provisioningNetworkCIDR")
+	if !found || err != nil {
+		glog.Errorf("provisioningNetworkCIDR not found in Baremetal provisioning CR %s", configName)
+		return BaremetalProvisioningConfig{}, err
+	}
+
+	// Checking for interdependecies within the config
+	if !foundDHCP && !foundNetwork {
+		glog.Errorf("Both the provisioningNetwork and provisioningDHCPExternal are not found in baremetal provisioning CR %s", configName)
+		return BaremetalProvisioningConfig{}, err
+	}
 
 	return BaremetalProvisioningConfig{
```

Comment:
I would have expected this struct to change to drop the `ProvisioningDHCPExternal` flag, since that is deprecated. Anything looking at that flag now needs to look at the value of `ProvisioningNetwork`, right?

Reply:
Since we are allowing ProvisioningDHCPExternal to be set in the provisioning CR, I did not remove it from BaremetalProvisioningConfig. But, since we are using that value to always set the ProvisioningNetwork, we can safely remove it.